### PR TITLE
Optimize SeekGE in committed iterator

### PR DESCRIPTION
**Pull request by johnnyaug**
When seeking multiple times in the same range, do not instantiate a new iterator to the underlying SSTable.

### CODE REVIEW

File: pkg/graveler/committed/iterator.go [hunk 1]
```diff
@@ -134,25 +134,30 @@ func (rvi *iterator) Close() {
...
 	rvi.rangesIt.SeekGE(Key(key))
-	if err = rvi.rangesIt.Err(); err != nil {
+	if err := rvi.rangesIt.Err(); err != nil {
 		rvi.err = err
-		return
+		return false
 	}
 	if !rvi.NextRange() {
-		return // Reached end.
+		return false // Reached end.3
```
Comment (guy-har) [inline pkg/graveler/committed/iterator.go:144]:
```suggestion
		return false // Reached end
```

**Review by guy-har — APPROVED** [review]
Verry nice optimization!!
Thanks!

**Review by arielshaqed — APPROVED** [review]
Neat.

Do you have a measurement of the number of calls to SeekGE it saves, or of some other calls saved / other speedup?

**Comment by johnnyaug** [comment]
@arielshaqed, we are going to post some more numbers today, but I will add something to the PR description for the record

**Comment by arielshaqed** [comment]
> @arielshaqed, we are going to post some more numbers today, but I will add something to the PR description for the record

Just to write it out in the largest font possible...

# 3x faster!

**Pull request merged**